src/tools/tidy/src/error_codes_check.rs

```diff
@@ -0,0 +1,183 @@
+//! Checks that all error codes have at least one test to prevent having error
+//! codes that are silently not thrown by the compiler anymore.
+
+use std::collections::HashMap;
+use std::io::{self, BufRead, BufReader};
+use std::fs::{self, DirEntry, File};
+use std::path::Path;
+
+const WHITELIST: &[&'static str] = &[
+    "E0183",
+    "E0227",
+    "E0279",
+    "E0280",
+    "E0311",
+    "E0313",
+    "E0314",
+    "E0315",
+    "E0377",
+    "E0456",
+    "E0461",
+    "E0462",
+    "E0464",
+    "E0465",
+    "E0472",
+    "E0473",
+    "E0474",
+    "E0475",
+    "E0476",
+    "E0479",
+    "E0480",
+    "E0481",
+    "E0482",
+    "E0483",
+    "E0484",
+    "E0485",
+    "E0486",
+    "E0487",
+    "E0488",
+    "E0489",
+    "E0514",
+    "E0519",
+    "E0523",
+    "E0526",
+    "E0554",
+    "E0570",
+    "E0629",
+    "E0630",
+    "E0640",
+    "E0717",
+    "E0727",
+    "E0729",
+];
+
+fn visit_dirs(dir: &Path, level: usize, cb: &mut dyn FnMut(&DirEntry)) -> io::Result<()> {
```

Comment:
There should be existing helpers in tidy that do this. For a good example look at src/tools/tidy/src/errors.rs.

Reply:
Indeed, code is much smaller now.